clinica/utils/inputs.py

```diff
@@ -0,0 +1,223 @@
+                     gives: ['/bids/sub-ADNI011S4105/ses-M00/pet/sub-ADNI011S4105_ses-M00_task-rest_acq-FDG_pet.nii.gz']
+
+            - Tricky example:
+                Get the file rh.white from FreeSurfer:
+                If you try:
+                    clinica_file_reader(['sub-ADNI011S4105'],
+                                        ['ses-M00'],
+                                        caps,
+                                        {'pattern': 'rh.white',
+                                         'description': 'right hemisphere of outter cortical surface.',
+                                         'needed_pipeline': 't1-freesurfer'})
+                        the following error will arise:
+                        * More than 1 file found::
+                            /caps/subjects/sub-ADNI011S4105/ses-M00/t1/freesurfer_cross_sectional/fsaverage/surf/rh.white
+                            /caps/subjects/sub-ADNI011S4105/ses-M00/t1/freesurfer_cross_sectional/rh.EC_average/surf/rh.white
+                            /caps/subjects/sub-ADNI011S4105/ses-M00/t1/freesurfer_cross_sectional/sub-ADNI011S4105_ses-M00/surf/rh.white
+                Correct usage (wanted for pet-surface): pattern string must be 'sub-*_ses-*/surf/rh.white')
+                    It then gives: ['/caps/subjects/sub-ADNI011S4105/ses-M00/t1/freesurfer_cross_sectional/sub-ADNI011S4105_ses-M00/surf/rh.white']
+
+        Note:
+            This function is case insensitive, meaning that the pattern argument can, for example, contain maj letter
+            that do not exists in the existing file path.
+
+    """
+
+    from os.path import join, isdir
+    from os import listdir
+    from clinica.utils.io import check_bids_folder, check_caps_folder
+    from colorama import Fore
+
+    def determine_caps_or_bids(input_dir):
+        """
+        Determines if the input is a CAPS or a BIDS folder
+        :param input_dir: input folder
+        :return: True if input_dir is a bids, False if input_dir is a CAPS
+
+            raise :
+                RuntimeError if function could not determine if BIDS or CAPS or whatever else
+        """
+        if isdir(join(input_directory, 'subjects')):
+            if len([sub for sub in listdir(join(input_dir, 'subjects')) if (sub.startswith('sub-') and isdir(join(input_dir, 'subjects', sub)))]) > 0 or isdir(join(input_dir, 'groups')):
+                return False
+            else:
+                raise RuntimeError('Could not determine if ' + input_dir + ' is a CAPS or BIDS directory')
+
+        else:
+            if len([sub for sub in listdir(input_dir) if (sub.startswith('sub-') and isdir(join(input_dir, sub)))]) > 0:
+                return True
+            else:
+                if isdir(join(input_dir, 'groups')):
+                    return False
+                else:
+                    raise RuntimeError('Could not determine if ' + input_dir + ' is a CAPS or BIDS directory')
```

Comment:
This can be extracted from `clinica_file_reader` and be used for other functions/methods :)

Reply:
BTW, you are not using Google Python Style for the docstrings.

Reply:
Done